### DISCUSSION TITLE
Save checkpoints without overwrite

### DISCUSSION THREAD

**Discussion by mcomunita** [Lightning-AI/pytorch-lightning#11087] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Hi there,

I am using the ModelCheckpoint callback to save my model every n epochs but I cannot find a way to prevent PL from overwriting/deleting the previous checkpoint.

Ideally, I would like to keep the default naming convention {epoch}-{step} but without losing previous checkpoints.

Thanks

**Comment by rohitgr7** [comment, marked as answer]
you can create a custom ModelCheckpoint instance with `save_top_k=-1` and pass it in inside Trainer callbacks.

```py
ckpt_callback = ModelCheckpoint(save_top_k=-1, ...)
trainer = Trainer(callbacks=[ckpt_callback], ...)
```

**Reply by mcomunita** [reply]
Thanks!
